internal/verifier/compare.go

```diff
@@ -173,21 +194,28 @@ func (verifier *Verifier) compareDocsFromChannels(
 			dstDoc = curDocWithTs
 		}
 
+		defer pool.Put(srcDoc.doc)
+		defer pool.Put(dstDoc.doc)
+
 		// Finally we compare the documents and save any mismatch report(s).
-		mismatches, err := verifier.compareOneDocument(srcDoc.doc, dstDoc.doc, namespace)
+		curProblems, err := verifier.compareOneDocument(srcDoc.doc, dstDoc.doc, namespace)
+
 		if err != nil {
 			return errors.Wrap(err, "failed to compare documents")
 		}
 
-		pool.Put(srcDoc.doc)
-		pool.Put(dstDoc.doc)
+		var mismatchTimes option.Option[recheck.MismatchTimes]
+		if len(curProblems) > 0 {
```

Comment:
It seems like if `len(curProblems) == 0` you could just short-circuit and return `nil` instead of checking it here.

Reply:
done